app/utils.py

```diff
@@ -1165,3 +1167,36 @@ def delete(self, *args):
     def trace(self, *args):
         """Default handler implementation which returns HTTP status 405."""
         return self.__return_unimplemented_method_error()
+
+
+# ==== XSRF protection =========================================================
+
+class XsrfTool(object):
+
+    # XSRF tokens expire after 4 hours.
+    TOKEN_EXPIRATION_TIME = 60 * 60 * 4
+
+    def __init__(self):
+        self._key = config.get('xsrf_token_key').encode('utf-8')
```

Comment:
Can you add initialization logic of this config somewhere so that new instances will have a valid key? e.g., Initialize it with a random string when it is not set.

Reply:
Yeah, good idea. Done with the tests in PR #557.